src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -622,4 +657,330 @@ export default class GroupComparisonStore extends ComparisonStore {
+        await: () => [
+            this._filteredAndAnnotatedMutationsReport,
+            this.sampleKeyToSample,
+        ],
+        invoke: () => {
+            const filteredMutations = compileMutations(
+                this._filteredAndAnnotatedMutationsReport.result!,
+                !this.driverAnnotationSettings.includeVUS,
+                !this.includeGermlineMutations
+            );
+            const filteredSampleKeyToSample = this.sampleKeyToSample.result!;
+            return Promise.resolve(
+                filteredMutations.filter(
+                    m => m.uniqueSampleKey in filteredSampleKeyToSample
+                )
+            );
+        },
+    });
+
+    readonly _filteredAndAnnotatedMutationsReport = remoteData({
+        await: () => [
+            this.mutations,
+            this.getMutationPutativeDriverInfo,
+            this.entrezGeneIdToGene,
+        ],
+        invoke: () => {
+            return Promise.resolve(
+                filterAndAnnotateMutations(
+                    this.mutations.result!,
+                    this.getMutationPutativeDriverInfo.result!,
+                    this.entrezGeneIdToGene.result!
+                )
+            );
+        },
+    });
+
+    readonly getMutationPutativeDriverInfo = remoteData({
+        await: () => {
+            const toAwait = [];
+            if (this.driverAnnotationSettings.oncoKb) {
+                toAwait.push(this.oncoKbMutationAnnotationForOncoprint);
+            }
+            if (this.driverAnnotationSettings.hotspots) {
+                toAwait.push(this.isHotspotForOncoprint);
+            }
+            if (this.driverAnnotationSettings.cbioportalCount) {
+                toAwait.push(this.getCBioportalCount);
+            }
+            if (this.driverAnnotationSettings.cosmicCount) {
+                toAwait.push(this.getCosmicCount);
+            }
+            return toAwait;
+        },
+        invoke: () => {
+            return Promise.resolve((mutation: Mutation): {
+                oncoKb: string;
+                hotspots: boolean;
+                cbioportalCount: boolean;
+                cosmicCount: boolean;
+                customDriverBinary: boolean;
+                customDriverTier?: string;
+            } => {
+                const getOncoKbMutationAnnotationForOncoprint = this
+                    .oncoKbMutationAnnotationForOncoprint.result!;
+                const oncoKbDatum:
+                    | IndicatorQueryResp
+                    | undefined
+                    | null
+                    | false =
+                    this.driverAnnotationSettings.oncoKb &&
+                    getOncoKbMutationAnnotationForOncoprint &&
+                    !(
+                        getOncoKbMutationAnnotationForOncoprint instanceof Error
+                    ) &&
+                    getOncoKbMutationAnnotationForOncoprint(mutation);
+
+                const isHotspotDriver =
```

Comment:
Are we using isHotspotDriver etc in this context or can we just kill all this code